build.sbt

```diff
@@ -556,6 +556,14 @@ lazy val sharedSettings = Def.settings(
     if (isScala213.value) List("-Ymacro-annotations")
     else Nil
   },
+  scalacOptions ++= {
+    if (isScala213.value)
+      List(
+        "-Xfatal-warnings",
+        "-Wconf:cat=deprecation:is,cat=other-match-analysis:i"
```

Comment:
@tgodzik @bjaglin perhaps you have some ideas; the `match-analysis` entry is only for "unreachable code" warnings coming from something in https://github.com/scalameta/scalameta/blob/main/scalameta/common/shared/src/main/scala/scala/meta/internal/transversers/transverser.scala (or the two adjacent implementations, but since both of them show that message, i assumed it was the parent).

since this is a macro, the message doesn't really help in figuring out which actual line of code leads to this. if you have some idea on how to find it, we could fix the problem -- and remove this extra warning exclusion.

Reply:
i figured it out: #3562.